xray/core/groupby.py

```diff
@@ -252,6 +252,27 @@ def fillna(self, value):
         """
         return self._fillna(value)
 
+    def where(self, cond):
```

Comment:
You could have something like a mask that varies by calendar month. e.g., to select values with less than 10 on average by calendar month:
`ds.groupby('time.month').where((ds < 10).groupby('time.month').mean('time'))`